js/DropZone/DropZoneValidatorDispatcher.js

```diff
@@ -0,0 +1,88 @@
+class DropZoneValidatorDispatcher {
+    /**
+     * DropZoneValidator
+     * @param {DropZoneValidationUtils} utils
+     * @param {DropZoneErrors} errors
+     * @param {Array} whitelist
+     * @param {number} maxFiles
+     * @param {number} maxSize
+     */
+    constructor (utils, errors, whitelist, maxFiles, maxSize) {
+        this.utils = utils;
+        this.errors = errors;
+        this.whitelist = whitelist;
+        this.maxFiles = maxFiles;
+        this.maxSize = maxSize;
+    }
+
+    /**
+     * Dispatch validation methods
+     * @param {Array} files
+     * @param {Number} totalFiles
+     * @param {Number} totalSize
+     * @returns {Object} error
+     */
+    validate (files, totalFiles, totalSize) {
+        let result = { valid: true, text: '' };
+        let fileCount = totalFiles;
+        let sizeCount = totalSize;
+
+        // if we have files but we do not have a length we're dealing
+        // with a browser with limited support, so we'll return them as valid
+        // now and let them be caught later
+        if (!files.length) {
+            return result;
+        }
+
+        for (let i = 0; i < files.length; i++) {
```

Comment:
yep good spot, Nikos pointed this one out also. Initially I was weary of needing the index, but I've been through and it's not required.